### PR TITLE
Added filter for masterwork kill tracker count.

### LINKED ISSUE #5624: masterwork enemies defeated filter
With sunsetting coming soon, many people will want to delete all the weapons that can't increase power further.

However there might be some who want to keep masterwork weapons that have high kill counts on them as trophies.

I'd like a filter like this:
is:masterwork powerlimit:=1060 enemiesdefeated:<1000

### PR DESCRIPTION
**Pull request by ryan-rushton**
Fixes #5624

### CODE REVIEW

**Review by nev-r — APPROVED** [review]

**Comment by nev-r** [comment]
nice

File: src/app/search/FilterHelp.tsx [hunk 1]
```diff
@@ -97,9 +97,10 @@ export default connect<StoreProps>(mapStateToProps)(FilterHelp);
 
 function FilterExplanation({ filter }: { filter: FilterDefinition }) {
   const dispatch = useDispatch();
+  const additionalSuggestions = filter.suggestionsGenerator?.() || [];
   const suggestions = Array.from(
     new Set(
-      generateSuggestionsForFilter(filter).filter(
+      [...generateSuggestionsForFilter(filter), ...additionalSuggestions].filter(
```
Comment (bhollis) [inline src/app/search/FilterHelp.tsx:103]: Does `generateSuggestionsForFilter` not already use `suggestionsGenerator`?

**Review by bhollis — APPROVED** [review]
Nice!

File: src/app/search/FilterHelp.tsx (hunk 1 above)
Comment (ryan-rushton) [reply src/app/search/FilterHelp.tsx:103]: No the other way around. Where we have created a `suggestionsGenerator ` we use `generateSuggestionsForFilter ` as the function with tweaked parameters.

Comment (nev-r) [reply src/app/search/FilterHelp.tsx:103]: i had it suggestionsgenerator running alongside the formulaic `format` based one, but i can see nesting them instead. i'll loop back and do that once this is settled.

**Comment by ryan-rushton** [comment]
Had to split the search term generation out into a utils file as there was a circular dependency jest didn't like when using `generateSuggestionsForFilter` inside an actual filter.

**Pull request merged**